src/ffi/ffi.rs

```diff
@@ -644,8 +644,8 @@ impl<'a> ArrowArrayRef for ArrowArrayChild<'a> {
         self.schema
     }
 
-    fn child(&self, _: usize) -> Result<ArrowArrayChild> {
-        todo!()
+    fn child(&self, index: usize) -> Result<ArrowArrayChild> {
+        create_child(self.array, self.schema, self.parent.clone(), index)
```

Comment:
Since the implementation is the same I think that you can keep just the one in ArrowArrayRef (this is how it's done in arrow-rs)

Reply:
Sorry, I am not sure I am following: what do you mean with "you can keep just the one in ArrowArrayRef"?

Reply:
There is a default implementation in the trait in https://github.com/jorgecarleitao/arrow2/blob/a5035bd016ad89121bc528d547bfbc9e74880799/src/ffi/ffi.rs#L568

In `impl ArrowArrayRef for Arc<ArrowArray>` you rely on it and don't reimplement child. Can't you eo the same here?

Reply:
you are right, even better. Thanks!